packages/database/src/changeHandlers/SurveyResponseOutdater.js

```diff
@@ -132,13 +132,13 @@ export class SurveyResponseOutdater extends ChangeHandler {
    * This method processes the changed records and ensures that all responses in the affected
    * dimension combos get a correct `outdated` status
    */
-  async handleChanges(changedResponses) {
-    const surveysById = await this.fetchSurveysById(changedResponses);
+  async handleChanges(transactionModels, changedResponses) {
```

Comment:
So this pattern of passing in `transactionModels` and passing it all over the place is kinda awkward huh?

I don't love it, it really makes things look really bloated. But I also don't love some of the other options that come to mind:
1. A `transactionModels` property on the class, which `ChangeHandler` sets prior to calling `handleChanges()` and removes afterwards. I don't like that it's unclear when `transactionModels` is set from the child class's perspective.
2. Similar to above but replacing the `models` property with the `transactionModels`. I don't like that it hides the reality of this work being done in a transaction (it's important for the developer to know imo)

Overall I think I do like this method most... but still really don't love it 😅 Any other ideas?

Reply:
Oh haha same thing as my last comment! I was wondering about capturing everything that happens with a transacting models in a class, which I think works in most cases?